internal/routers/routers.go

```diff
@@ -29,10 +29,22 @@ func NewAPIRouter(
 	clientIdCli string,
 	oidcURL string,
 	oidcBackchannel string,
-	insecureTLS bool) (*gin.Engine, error) {
+	insecureTLS bool,
+	browserFlow *agent.OidcAgent,
+	deviceFlow *agent.OidcAgent,
+) (*gin.Engine, error) {
+
 	gin.SetMode(gin.ReleaseMode)
 	r := gin.New()
 
+	loggerMiddleware := ginzap.GinzapWithConfig(logger.Desugar(), &ginzap.Config{TimeFormat: time.RFC3339, UTC: true, TraceID: true})
+	r.Use(otelgin.Middleware(name))
+	r.Use(ginzap.RecoveryWithZap(logger.Desugar(), true))
+
+	r.Use(browserFlow.CookieSessionMiddleware())
+	agent.AddCodeFlowRoutes(r.Group("/browser", loggerMiddleware), browserFlow)
+	agent.AddDeviceFlowRoutes(r.Group("/device", loggerMiddleware), deviceFlow)
```

Comment:
We should probably look at the API here.
It would make more sense to me to do something like this:
```go
webAuthProxy := r.Group("/web/", loggerMiddleware)
webAuthProxy.Use(agent.NewRouter( ...)
```

As it stands the API names are incorrect. One doesn't do code flow, and the other device flow.
They both do whatever flows are configured for the client-id/secret.

Reply:
Note: I blame myself for those API names 😆

Reply:
I don't mind the path rename or method refactors.  Could we do it in a follow up PR?